tests/phpunit/tests/rest-api/rest-themes-controller.php

```diff
@@ -127,6 +127,9 @@ public static function wpTearDownAfterClass() {
 		self::delete_user( self::$subscriber_id );
 		self::delete_user( self::$contributor_id );
 		self::delete_user( self::$admin_id );
+
+		remove_theme_support( 'editor-gradient-presets' );
+		remove_theme_support( 'editor-color-palette' );
```

Comment:
I was unaware of this until now, but added theme support for a feature in a test will persist in other tests even across theme switches, unless explicitly removed. 

The `remove_theme_support( 'editor-color-palette' );` in the teardown function of `WpGetGlobalStylesheet` below was causing a failure in one of the `wpThemeJsonResolver` tests, because the theme support for `'editor-gradient-presets'` added in this test was changing the order of the color settings output from theme.json, which gets merged with theme supports [here](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/class-wp-theme-json-resolver.php#L332). I decided to remove them both explicitly in this test to avoid future issues.

Reply:
Does it work if we call `remove_theme_support` at the end of the test?

So like this:

```php
	public function test_theme_supports_editor_gradient_presets_array() {
		$gradient = array(
			'name'     => __( 'Vivid cyan blue to vivid purple', 'themeLangDomain' ),
			'gradient' => 'linear-gradient(135deg,rgba(6,147,227,1) 0%,rgb(155,81,224) 100%)',
			'slug'     => 'vivid-cyan-blue-to-vivid-purple',
		);
		add_theme_support( 'editor-gradient-presets', array( $gradient ) );
		$response = self::perform_active_theme_request();
		$result   = $response->get_data();
		$this->assertArrayHasKey( 'theme_supports', $result[0] );
		$this->assertSame( array( $gradient ), $result[0]['theme_supports']['editor-gradient-presets'] );
		remove_theme_support( 'editor-gradient-presets' );
	}
```

No big deal, happy with whatever works, it's just that usually `wpTearDownAfterClass` only contains the "mirror image" of what's in `set_up` or `wpSetUpBeforeClass`.

Reply:
It should work! Lemme try that

Reply:
Just a question, what happens in PHPUnit if you call a function after a failing assertion? Would `remove_theme_support` still get called? If not, then we might want to have `remove_theme_support` be called just before the first assertion so that the absence of calling `remove_theme_support` doesn't affect other tests 🤔

My main thinking there is so that if this test "really" fails, and someone's trying to investigate, they don't get distracted by another failing test that's only failing if this test didn't manage to do all the cleanup needed.